fvm/src/kernel/default.rs

```diff
@@ -876,11 +743,7 @@ where
             .create_actor(code_id, actor_id, delegated_address)
     }
 
-    fn upgrade_actor<K: Kernel>(
-        &mut self,
-        new_code_cid: Cid,
-        params_id: BlockId,
-    ) -> Result<CallResult> {
+    fn upgrade_actor(&mut self, new_code_cid: Cid, params_id: BlockId) -> Result<CallResult> {
```

Comment:
Ah, I'd keep this like this and pass `K` to `call_actor`. It makes testing easier.

Reply:
Done, but moved to the Kernel trait due to ambassador errors which I could not get working